addons/event/models/event_registration.py

```diff
@@ -51,6 +51,15 @@ class EventRegistration(models.Model):
         ('open', 'Confirmed'), ('done', 'Attended')],
         string='Status', default='draft', readonly=True, copy=False, tracking=True)
 
+    def default_get(self, fields):
+        ret_vals = super().default_get(fields)
+        utm_defaults = self.env['utm.mixin'].default_get([field.replace('utm_', '') for field in fields])
```

Comment:
Sugg:

```py
utm_fields = ["source_id", "medium_id", "campaign_id"]
if not any([f'utm_{utm_field}' in fields for utm_field in utm_fields]):
   return ret_vals

utm_defaults = self.env['utm.mixin'].default_get(utm_fields)
for utm_field in utm_fields:
    event_utm_field = f'utm_{utm_field}'
    if event_utm_field  in fields and utm_defaults.get(utm_field):
        ret_vals[event_utm_field] = utm_defaults[utm_field]
```

That way we avoid unnecessary processing if the UTMs are not in the list of asked fields.

Reply:
So... you did not like the rest of my suggestion code? Or maybe you missed it? 😄 

I find it a bit easier to play with the hardcoded list rather than replacing dynamically by stripping "utm_" and also calling "tracking_fields" to only use one value, don't you think?

Reply:
"I hated it :angry:"
But actually I thought it was just for context and you hadn't changed it :see_no_evil: 

Indeed went with that, no need to bother with tracking_fields and such. I just hardcoded both lists in the end, doing string manipulations multiple times.